### PR TITLE
bootstrap: add logging

### PR DESCRIPTION
**Pull request by Williangalvani**
This moves bootstrap to loguru, and populates a file at the same place our other logs are

### CODE REVIEW

File: bootstrap/main.py [hunk 1]
```diff
@@ -4,15 +4,19 @@
 import sys
 
 import docker
-
+from loguru import logger
 from bootstrap.bootstrap import Bootstrapper
 
+
 if __name__ == "__main__":
+    logger.add("/var/logs/blueos/services/bootstrap/bootstrap.log", level="DEBUG", rotation="1 MB")
```
Comment (patrickelectric) [inline bootstrap/main.py:12]: Will this work if BlueOS rotator runs ?

Comment (Williangalvani) [reply bootstrap/main.py:12]: I have no idea. should I just take this out? I guess the other services don't any rotation by themselves, right?

Comment (patrickelectric) [reply bootstrap/main.py:12]: Let it run and check if there is a rotation and it's working fine, right now we have a default rotation of 10min:
https://github.com/bluerobotics/BlueOS/blob/master/core/libs/commonwealth/commonwealth/utils/logs.py#L63

And a zipper every hour:
https://github.com/bluerobotics/BlueOS/blob/master/core/start-blueos-core#L78

**Comment by patrickelectric** [comment]
I just tested the last version and it appears to be not working:

image

**Comment by patrickelectric** [comment]
> I just tested the last version and it appears to be not working:
> 
> image

How is the rotation is done exactly in this case ? For log_zipper, it'll zip the rotated logs.

**Comment by patrickelectric** [comment]
> > > I just tested the last version and it appears to be not working:
> > > image
> > 
> > 
> > How is the rotation is done exactly in this case ? For log_zipper, it'll zip the rotated logs older than X, but this file from what I can see is never rotated.
> 
> it relies on log_zipper, too. but that is running on 1 hour intervals, apparently.

My point is, if log_zipper runs, bootstrap will not generate new logs.

**Comment by Williangalvani** [comment]
> I just tested the last version and it appears to be not working:
> 

I was told we don't delete files while in use. There doesn't seem to be anything within loguru to deal with that.

I updated the code so it rotated every 10min, and the files are now timestamped, so they won't get reused.

**Comment by joaoantoniocardoso** [comment]
> > I just tested the last version and it appears to be not working:
> 
> I was told we don't delete files while in use. There doesn't seem to be anything within loguru to deal with that.
> 
> I updated the code so it rotated every 10min, and the files are now timestamped, so they won't get reused.

That's true, all the other services have the same problem.

**Comment by patrickelectric** [comment]
> > I just tested the last version and it appears to be not working:
> 
> I was told we don't delete files while in use. There doesn't seem to be anything within loguru to deal with that.

As pointed here https://github.com/bluerobotics/BlueOS/pull/1950#discussion_r1287459469.

In core, we rotate logs every 10 minutes, so the log being used will be T-10min.
log_zipper rotates logs older than 1h, So T-60min.
We don't zip open log files because they are too new to be zipped.




> 
> I updated the code so it rotated every 10min, and the files are now timestamped, so they won't get reused.

**Comment by Williangalvani** [comment]
> > > I just tested the last version and it appears to be not working:
> > 
> > 
> > I was told we don't delete files while in use. There doesn't seem to be anything within loguru to deal with that.
> 
> As pointed here [#1950 (comment)](https://github.com/bluerobotics/BlueOS/pull/1950#discussion_r1287459469).
> 
> In core, we rotate logs every 10 minutes, so the log being used will be T-10min. log_zipper rotates logs older than 1h, So T-60min. We don't zip open log files because they are too new to be zipped.
> 
> > I updated the code so it rotated every 10min, and the files are now timestamped, so they won't get reused.
> 
> The codes says 10MB not 10min. https://github.com/bluerobotics/BlueOS/pull/1950/files#diff-4d1dd542128fe2565a2688dc51e6e6bcc45fea5d94a4c4a476ef917fdd4c278fR12

lol, I mean't MB but wrote minutes. I think this will result in 1 log per boot, which should be perfectly fine

**Comment by patrickelectric** [comment]
> > > > I just tested the last version and it appears to be not working:
> > > 
> > > 
> > > I was told we don't delete files while in use. There doesn't seem to be anything within loguru to deal with that.
> > 
> > 
> > As pointed here [#1950 (comment)](https://github.com/bluerobotics/BlueOS/pull/1950#discussion_r1287459469).
> > In core, we rotate logs every 10 minutes, so the log being used will be T-10min. log_zipper rotates logs older than 1h, So T-60min. We don't zip open log files because they are too new to be zipped.
> > > I updated the code so it rotated every 10min, and the files are now timestamped, so they won't get reused.
> > 
> > 
> > The codes says 10MB not 10min. [#1950 (files)](https://github.com/bluerobotics/BlueOS/pull/1950/files#diff-4d1dd542128fe2565a2688dc51e6e6bcc45fea5d94a4c4a476ef917fdd4c278fR12)
> 
> lol, I mean't MB but wrote minutes. I think this will result in 1 log per boot, which should be perfectly fine

As said, if log_zipper runs, the bootstrap log file will be zipped, and bootstrap will not log anymore.

image
This is after log_zipper runs.

**Comment by Williangalvani** [comment]
how's this being handled by the other services?

**Comment by patrickelectric** [comment]
> how's this being handled by the other services?

https://github.com/bluerobotics/BlueOS/pull/1950#issuecomment-1673734257

**Comment by Williangalvani** [comment]
> > how's this being handled by the other services?
> 
> [#1950 (comment)](https://github.com/bluerobotics/BlueOS/pull/1950#issuecomment-1673734257)

The longer answer is "We initiate loguru with a [custom LogRotator](https://github.com/bluerobotics/BlueOS/blob/master/core/libs/commonwealth/commonwealth/utils/logs.py#L11). That makes Loguru itself rotate the files every 10 minutes. this way, the open file can never be older than 10 minutes.

When log_zipper runs, it looks for [files last modified earlier than 60 minutes ago](https://github.com/bluerobotics/BlueOS/blob/master/core/services/log_zipper/main.py#L56).

This means setting Loguru to 60 minutes should work, right? so we don't need to add commonwealth to Bootstrap.

**Comment by patrickelectric** [comment]
> > > how's this being handled by the other services?
> > 
> > 
> > [#1950 (comment)](https://github.com/bluerobotics/BlueOS/pull/1950#issuecomment-1673734257)
> 
> The longer answer is "We initiate loguru with a [custom LogRotator](https://github.com/bluerobotics/BlueOS/blob/master/core/libs/commonwealth/commonwealth/utils/logs.py?rgh-link-date=2023-08-10T19%3A12%3A09Z#L11). That makes Loguru itself rotate the files every 10 minutes. this way, the open file can never be older than 10 minutes.
> 
> When log_zipper runs, it looks for [files last modified earlier than 60 minutes ago](https://github.com/bluerobotics/BlueOS/blob/master/core/services/log_zipper/main.py?rgh-link-date=2023-08-10T19%3A12%3A09Z#L56).
> 
> This means setting Loguru to 60 minutes should work, right? so we don't need to add commonwealth to Bootstrap.

anything less than 60min should work, I would recommend to use between 30min and 10min.

**Review by patrickelectric — COMMENTED** [review]
tested and working

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**